### ISSUE TITLE
display-plugin-updates should optionally show plugins updates for unbound plugins

### ISSUE THREAD

**Issue by kwin** [mojohaus/versions#533]
Currently the goal `display-plugin-updates` only shows updates for plugins bound in the current project's lifecycle and plugins listed in the effective model in build->plugins->plugin (see https://github.com/mojohaus/versions-maven-plugin/blob/c0589f4a69afc1a53d8f2d461f621339fd0c0710/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L380).
In some cases one is really interested to see all possible plugin updates, even for the ones which are neither bound (e.g. because the project is of type "pom") nor enabled or ones only listed in some profiles.

A common use case is to check a parent pom for possible plugin updates. That will currently never emit infos for updates of e.g. "maven-jar-plugin" unless that plugin is explicitly mentioned (due to its packaging "pom").

**Comment by sultan** [comment]
i was thinking about having html reports updates for transitive dependencies, do you think you could implement your proposal for html too ? maybe one day vulnerabilities would be nice too.

**Comment by slawekjaranowski** [comment]
can you prepare a simple project which can be added to IT tests.

**Comment by ajarmoniuk** [comment]
I have been looking into this, trying to get to the root cause of the issue. I managed to reproduce the problem with https://github.com/apache/sling-parent/tree/sling-parent-reactor-47 and plugin version 2.8.1 through 2.9.0 after which #550 effectively hid the issue.
 
However, the problem seems to have been helped with with the latest snapshot (if I remove Stefan's patches):

```
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   maven-jar-plugin ................................... 3.2.2 -> 3.3.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
```

It appears that it's already been fixed at 2.10.0 with the patch reverted!

```
[INFO] --- versions-maven-plugin:2.10.0:display-plugin-updates (default-cli) @ sling-bundle-parent ---
[INFO] 
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   maven-jar-plugin ................................... 3.2.2 -> 3.3.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
[INFO] 
[INFO] All plugins have a version specified.
[INFO] 
[INFO] Project requires minimum Maven version for build of: 3.3.9
[INFO] Plugins require minimum Maven version of: 3.3.9
```

But it was of course there with 2.9.0:

```
[INFO] --- versions-maven-plugin:2.9.0:display-plugin-updates (default-cli) @ sling-bundle-parent ---
[INFO] 
[INFO] The following plugin updates are available:
[INFO]   biz.aQute.bnd:bnd-baseline-maven-plugin ............ 6.1.0 -> 6.4.0
[INFO]   biz.aQute.bnd:bnd-maven-plugin ..................... 6.1.0 -> 6.4.0
[INFO]   org.apache.sling:sling-maven-plugin ................ 2.4.2 -> 3.0.2
[INFO] 
[WARNING] The following plugins do not have their version specified:
[WARNING]   maven-jar-plugin ......................................... 3.3.0
[INFO] 
[INFO] Project requires minimum Maven version for build of: 3.3.9
[INFO] Plugins require minimum Maven version of: 3.3.9
[INFO] 
[INFO] No plugins require a newer version of Maven than specified by the pom.
```

It's highly possible that the issue might already have been resolved when #550 was merged.

It turns out that the root cause was different:

(https://github.com/mojohaus/versions/blob/versions-maven-plugin-2.9.0/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L530)

Here, the version, which was not specified in the current POM, but which was resolved from parents was being overwritten by the one from superpom. Which, in case of some plugins, is not specified. Hence, a resolved version became null, which was reported in the faulty report.
